client/asset/btc/spv.go

```diff
@@ -0,0 +1,1631 @@
+func (w *spvWallet) searchBlockForRedemptions(ctx context.Context, reqs map[outPoint]*findRedemptionReq,
+	blockHash chainhash.Hash) (discovered map[outPoint]*findRedemptionResult) {
+
+	// Just match all the scripts together.
+	scripts := make([][]byte, 0, len(reqs))
+	for _, req := range reqs {
+		scripts = append(scripts, req.pkScript)
+	}
+
+	discovered = make(map[outPoint]*findRedemptionResult, len(reqs))
+
+	matchFound, err := w.matchPkScript(&blockHash, scripts)
+	if err != nil {
+		w.log.Errorf("matchPkScript error: %w", err)
+		return
+	}
+
+	if !matchFound {
+		return
+	}
+
+	// There is at least one match. Pull the block.
+	block, err := w.cl.GetBlock(blockHash)
+	if err != nil {
+		w.log.Errorf("neutrino GetBlock error: %v", err)
+		return
+	}
+
+	for _, msgTx := range block.MsgBlock().Transactions {
+		newlyDiscovered := findRedemptionsInTx(ctx, true, reqs, msgTx, w.chainParams)
+		for outPt, res := range newlyDiscovered {
+			discovered[outPt] = res
+		}
+	}
+	return
+}
+
+// findRedemptionsInMempool is unsupported for SPV.
+func (w *spvWallet) findRedemptionsInMempool(ctx context.Context, reqs map[outPoint]*findRedemptionReq) (discovered map[outPoint]*findRedemptionResult) {
+	return
+}
+
+// confirmations looks for the confirmation count and spend status on a
+// transaction output that pays to this wallet.
+func (w *spvWallet) confirmations(txHash *chainhash.Hash, vout uint32) (blockHash *chainhash.Hash, confs uint32, spent bool, err error) {
+	details, err := w.wallet.walletTransaction(txHash)
+	if err != nil {
+		return nil, 0, false, err
+	}
+	if details == nil {
+		return nil, 0, false, WalletTransactionNotFound
+	}
+
+	if details.Block.Hash != (chainhash.Hash{}) {
+		blockHash = &details.Block.Hash
+	}
+
+	for _, credit := range details.Credits {
+		if credit.Index == vout {
+			syncBlock := w.wallet.syncedTo() // Better than chainClient.GetBestBlockHeight() ?
+			return blockHash, uint32(confirms(details.Block.Height, syncBlock.Height)), credit.Spent, nil
+		}
+	}
+
+	return blockHash, 0, false, SpentStatusUnknown
+}
+
+// getTransaction retrieves the specified wallet-related transaction.
+// This is pretty much a copy-past from btcwallet 'gettransaction' JSON-RPC
+// handler.
+func (w *spvWallet) getTransaction(txHash *chainhash.Hash) (*GetTransactionResult, error) {
+	// Option # 1 just copies from UnstableAPI.TxDetails. Duplicating the
+	// unexported bucket key feels dirty.
+	//
+	// var details *wtxmgr.TxDetails
+	// err := walletdb.View(w.Database(), func(dbtx walletdb.ReadTx) error {
+	// 	txKey := []byte("wtxmgr")
+	// 	txmgrNs := dbtx.ReadBucket(txKey)
+	// 	var err error
+	// 	details, err = w.TxStore.TxDetails(txmgrNs, txHash)
+	// 	return err
+	// })
+
+	// Option #2
+	// This is what the JSON-RPC does (and has since at least May 2018).
+	details, err := w.wallet.walletTransaction(txHash)
+	if err != nil {
+		return nil, err
+	}
+	if details == nil {
+		return nil, WalletTransactionNotFound
+	}
```

Comment:
It looks like `w.wallet.walletTransaction` is already checking for this, but fine to check again, as wallet is an interface I guess.